### ISSUE TITLE
Restore-dbfromfilteredarray not account for filename with multiple periods

### ISSUE THREAD

**Issue by kclark1117** [sqlcollaborative/dbatools#2144]
### General Troubleshooting steps

- [x] Verified running latest release of dbatools?
- [x] Verified errors are not related to permissions?
- [x] Can duplicate in new/fresh PowerShell.exe session?

### Version Information
 - Operating System (Name|Version): WindowsServer2012R2
 - PowerShell Version: 5 

## Steps to Reproduce

create a database which has a filename that includes multiple periods. Such as NewTestV1.2.1.mdf and then copy that database with copy-dbadatabase. When the command is complete look at the new filename of the copied database.

### Problem to solve
We have a number of Databases which include additional periods (.) in the physical filenames. When using copy-dbadatabase with the -BackupRestore switch, the physical filenames are being set incorrectly at the new server. i.e.
NewDatabase_2017.1.mdf would become: "NewDatabase_2017.1 mdf"

### Additional information
 It appears that restore-dbfromfilteredarray is assuming the filename only contains a single period to denote the extension and is using the following to attempt to parse the value:
`$filename, $extension = (Split-Path $file.PhysicalName -leaf).split('.')`

some additional logic to test if the split returns more than two values would be helpful. In my local restore-dbfromfilteredarray I replaced the above line with the following which resolved things, though I'm sure there's a more eloquent solution:
```
$tempfname = (Split-path $file.PhysicalName -leaf).split('.')
if($tempfname.count -gt 2){
    $filename = $tempfname[0..($tempfname.length-2)] -join '.'
    $extension = $tempfname[$tempfname.length-1]
    }
else{
    $filename,$extension = $tempfname
}
```

**Comment by niphlod** [comment]
There are definitely better and "powershell core" methods to get the name
and the extension of a file. If @stuart-moore doesn't chime in, I'll submit
a fix within the day.

Thanks for the report!

Il 24 ago 2017 5:49 AM, "kclark1117" <notifications@github.com> ha scritto:

**Comment by Stuart-Moore** [comment]
Going to go with

$Extension = ($file.PhysicalName.split('.'))[-1]
$Filename  = $file.PhysicalName.replace(".$extension","")

Better options accepted, but most of the FSO based ones don't work as that
file doesn't always exist as it's being pulled from the headers. Could
create an FSO object i guess if that'd be preferred?

**Comment by niphlod** [comment]
@Stuart-Moore : io.path comes to rescue
```
[IO.Path]::GetFileNameWithoutExtension('nonexistent.dbatools.fullpower')
[IO.Path]::GetExtension('nonexistent.dbatools.fullpower')
```

**Comment by Stuart-Moore** [comment]
Fixed in #2145
